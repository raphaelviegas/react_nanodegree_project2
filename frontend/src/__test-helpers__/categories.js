export const initialState = { categories: [] }

export const categories = [
    {
        name: 'react',
        path: 'react'
    },
    {
        name: 'redux',
        path: 'redux'
    },
    {
        name: 'udacity',
        path: 'udacity'
    }
]