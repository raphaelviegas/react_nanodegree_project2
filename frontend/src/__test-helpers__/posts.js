const uuidv4 = require('uuid/v4')

export const initialState = {
    posts: [],
    sortByOption: '-voteScore',
    invalidPost: false,
    loadingPosts: true,
}

export const posts = [
    {
        id: '8xf0y6ziyjabvozdd253nd',
        timestamp: 1467166872634,
        title: 'Udacity is the best place to learn React',
        body: 'Everyone says so after all.',
        author: 'thingtwo',
        category: 'react',
        voteScore: 10,
        deleted: false,
        commentCount: 2
    },
    {
        id: '6ni6ok3ym7mf1p33lnez',
        timestamp: 1468479767190,
        title: 'Learn Redux in 10 minutes!',
        body: 'Just kidding. It takes more than 10 minutes to learn technology.',
        author: 'thingone',
        category: 'redux',
        voteScore: -5,
        deleted: false,
        commentCount: 0
    }
]

export const removePost = '8xf0y6ziyjabvozdd253nd'

export const votePost = {
    id: '8xf0y6ziyjabvozdd253nd',
    voteScore: 10
}

let id = uuidv4()
export const addPostInfo = {
    author: "Test Author",
    body: "Test Body",
    category: "Test",
    title: "Test Title",
    timestamp: Date.now(),
    id
}

export const editPostInfo = {
    id: '8xf0y6ziyjabvozdd253nd',
    editContent: {
        title: 'Edited post title',
        body: 'Edited post body'
    }
}