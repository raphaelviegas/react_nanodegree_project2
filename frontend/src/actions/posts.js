import * as ServerAPI from '../helpers/ServerAPI'

export const IS_LOADING = 'IS_LOADING'
export const LOAD_POSTS = 'LOAD_POSTS'
export const REMOVE_POST = 'REMOVE_POST'
export const VOTE_POST = 'VOTE_POST'
export const ORDER_POSTS = 'ORDER_POSTS'
export const ADD_POST = 'ADD_POST'
export const EDIT_POST = 'EDIT_POST'

//Local state
export function isLoading() {
  return {
    type: IS_LOADING
  }
}

export function loadPosts(posts) {
  return {
    type: LOAD_POSTS,
    payload: posts
  }
}

export function votePost(id, voteScore) {
  return {
    type: VOTE_POST,
    payload: { id, voteScore }
  }
}

export function removePost(id) {
  return {
    type: REMOVE_POST,
    payload: id
  }
}

export function orderPosts(orderOption) {
  return {
    type: ORDER_POSTS,
    payload: orderOption
  }
}

export function addPost(post) {
  return {
    type: ADD_POST,
    payload: post
  }
}

export function editPost(id, editContent) {
  return {
    type: EDIT_POST,
    payload: {
      id,
      editContent
    }
  }
}

//Fetch requests
export const fetchPosts = () => dispatch => (
  ServerAPI.getPosts()
    .then(posts => dispatch(loadPosts(posts)))
)

export const fetchVotePost = (id, option) => dispatch => (
  ServerAPI.vote('posts', id, option)
    .then(post => dispatch(votePost(post.id, post.voteScore)))
)

export const fetchRemovePost = (id) => dispatch => (
  ServerAPI.remove('posts', id)
    .then(post => dispatch(removePost(post.id)))
)

export const fetchAddPost = (postInfo) => dispatch => (
  ServerAPI.add('posts', postInfo)
    .then(post => dispatch(addPost(post)))
)

export const fetchEditPost = (id, editContent) => dispatch => (
  ServerAPI.edit('posts', id, editContent)
    .then(() => dispatch(editPost(id, editContent)))
)





