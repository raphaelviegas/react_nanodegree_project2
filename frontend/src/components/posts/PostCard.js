import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import PropTypes from 'prop-types'
import MaterialIcon from 'material-icons-react'
import VoteScore from '../shared/VoteScore'
import EditPostForm from '../posts/editPost/EditPostForm'

class PostCard extends Component {
    static propTypes = {
        post: PropTypes.object.isRequired,
        onRemovePost: PropTypes.func.isRequired
    }

    state = {
        showEdit: '',
    }

    showEdit = () => this.setState(() => ({ showEdit: 'card-reveal-active' }))
    hideEdit = () => this.setState(() => ({ showEdit: '' }))

    removePost(id) {
        this.props.onRemovePost(id)
    }
    render() {
        const { id, title, body, author, voteScore, commentCount, category } = this.props.post
        const { showEdit } = this.state
        return (
            <div className="col s12 m6">
                <div className="card grey lighten-4">
                    <div className="card-content">
                        <span className="card-title truncate">{title}</span>
                        <span className="card-author">{author}</span>
                        <p>{body}</p>
                    </div>
                    <div className="card-action valign-wrapper">
                        <VoteScore size={16} score={voteScore} type='post' id={id} />
                        <Link className="align-right" to={`/${category}/${id}`}>Show Post</Link>
                    </div>
                    <div className="card-comments">
                        <span className="card-comments-number">{commentCount}</span>
                    </div>
                    <div className="card-buttons">
                        <div className="card-button-edit" onClick={() => this.showEdit()}>
                            <MaterialIcon icon="edit" size={14} />
                        </div>
                        <div className="card-button-delete" onClick={() => this.removePost(id)}>
                            <MaterialIcon icon="delete" size={14} />
                        </div>
                    </div>
                    <div className={`card-reveal ${showEdit}`}>
                        <EditPostForm
                            id={id}
                            defaultTitle={title}
                            defaultBody={body}
                            onPostCardFormSubmited={this.hideEdit}
                        />
                        <div className="hide-edit" onClick={() => this.hideEdit()}>
                            <MaterialIcon icon="close" color="#c3c3c3" />
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}

export default PostCard
