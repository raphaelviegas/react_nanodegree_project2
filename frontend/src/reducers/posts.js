import {
    IS_LOADING,
    LOAD_POSTS,
    REMOVE_POST,
    VOTE_POST,
    ORDER_POSTS,
    ADD_POST,
    EDIT_POST
} from '../actions/posts'

function postReducer(state = {
    posts: [],
    sortByOption: '-voteScore',
    invalidPost: false,
    loadingPosts: true,
}, action) {
    switch (action.type) {
        case IS_LOADING: {
            state = {
                ...state,
                loadingPosts: true
            }
            return state
        }
        case LOAD_POSTS:
            state = {
                ...state,
                loadingPosts: false,
                posts: action.payload
            }
            return state
        case REMOVE_POST:
            state = {
                ...state,
                posts: state.posts.filter(post => post.id !== action.payload)
            }
            return state
        case VOTE_POST:
            state = {
                ...state,
                posts: state.posts.map(post => {
                    if (post.id === action.payload.id) {
                        return {
                            ...post,
                            voteScore: action.payload.voteScore
                        }
                    }
                    return post
                }),
            }
            return state
        case ORDER_POSTS:
            state = {
                ...state,
                sortByOption: action.payload
            }
            return state
        case ADD_POST:
            let posts = state.posts.concat([action.payload])
            state = {
                ...state,
                loadingPosts: false,
                posts
            }
            return state
        case EDIT_POST:
            state = {
                ...state,
                posts: state.posts.map(post => {
                    if (post.id === action.payload.id) {
                        post.title = action.payload.editContent.title
                        post.body = action.payload.editContent.body
                    }
                    return post
                })
            }
            return state
        default:
            return state
    }
}

export default postReducer