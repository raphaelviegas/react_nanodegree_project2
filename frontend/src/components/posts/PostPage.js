import React, { Component } from 'react'
import { connect } from 'react-redux'
import { withRouter, Redirect } from 'react-router-dom'
import { fetchPosts, fetchRemovePost } from '../../actions/posts'
import MaterialIcon from 'material-icons-react'
import Moment from 'react-moment';
import VoteScore from '../shared/VoteScore'
import Comments from '../comments/Comments'
import EditPostModal from '../posts/editPost/EditPostModal'

export class PostPage extends Component {
    state = {
        modalOpen: ''
    }

    openModal = () => this.setState(() => ({ modalOpen: 'modal-open' }))
    closeModal = () => this.setState(() => ({ modalOpen: '' }))

    componentDidMount() {
        this.props.fetchPosts()
    }

    componentDidUpdate(prevProps) {
        if (this.props.id !== prevProps.id) {
            this.props.fetchPosts()
        }
    }

    removePost = (id) => {
        this.props.fetchRemovePost(id)
        this.props.history.push('/')
    }

    render() {
        const { postInfo } = this.props
        let invalidPost = (postInfo.length > 0) ? false : true
        return (
            <div>
                {invalidPost
                    ? <Redirect to="/error" />
                    : <div className="post-page">
                        {postInfo.map((post) => (
                            <div key={post.id}>
                                <div className="post-buttons">
                                    <div className="post-button-edit" onClick={() => this.openModal()}>
                                        <MaterialIcon icon="edit" size={20} />
                                    </div>
                                    <div className="post-button-remove" onClick={() => this.removePost(post.id)}>
                                        <MaterialIcon icon="delete" size={20} />
                                    </div>
                                </div>
                                <div className="container">
                                    <h4 className="post-title">{post.title}</h4>
                                    <div className="post-infos">
                                        <span className="post-author">Author: {post.author}</span>
                                        <span className="post-category">Category: {post.category}</span>
                                        <span className="post-date">
                                            <MaterialIcon icon="calendar_today" size={16} />
                                            <Moment format="DD/MMM/YY">{post.timestamp}</Moment>
                                        </span>
                                    </div>
                                    <VoteScore size={18} score={post.voteScore} type='post' id={post.id} />
                                    <p className="post-body">{post.body}</p>
                                    <Comments parentId={post.id} />
                                </div>
                                <EditPostModal
                                    modalOpen={this.state.modalOpen}
                                    onCloseModal={this.closeModal}
                                    id={post.id}
                                    title={post.title}
                                    body={post.body}
                                />
                            </div>
                        ))}
                    </div>
                }
            </div>
        )
    }
}
const mapStateToProps = (state, ownProps) => {
    return {
        id: ownProps.match.params.postId,
        postInfo: state.postsReducer.posts.filter(post => post.id === ownProps.match.params.postId)
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        fetchPosts: () => dispatch(fetchPosts()),
        fetchRemovePost: (id) => dispatch(fetchRemovePost(id))
    }
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(PostPage));
