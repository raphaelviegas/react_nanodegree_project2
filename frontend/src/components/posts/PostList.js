import React, { Component } from 'react'
import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'
import PropTypes from 'prop-types'
import { fetchRemovePost } from '../../actions/posts'
import PostCard from './PostCard'
import sortBy from 'sort-by'

export class PostList extends Component {
  static propTypes = {
    posts: PropTypes.array.isRequired,
    sortByOption: PropTypes.string.isRequired
  }

  removePost = (id) => {
    this.props.fetchRemovePost(id)
  }

  render() {
    const { posts, sortByOption } = this.props
    //SortBy
    const sortedPosts = [...posts].sort(sortBy(sortByOption))
    return (
      <div>
        {posts.length === 0
          ? <div className="noPosts"><h4>No posts to show</h4></div>
          : <div className="row">
            {
              sortedPosts.map((post, index) => {
                return <PostCard post={post} onRemovePost={this.removePost} key={index} />
              })
            }
          </div>
        }
      </div>
    )
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    posts: ownProps.posts,
    sortByOption: state.postsReducer.sortByOption
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    fetchRemovePost: (id) => dispatch(fetchRemovePost(id))
  }
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(PostList));
