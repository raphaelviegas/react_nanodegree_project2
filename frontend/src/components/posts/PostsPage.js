import React, { Component } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import { fetchPosts } from '../../actions/posts'
import PostList from '../posts/PostList'
import OrderPosts from '../shared/OrderPosts'

export class PostsPage extends Component {
    static propTypes = {
        posts: PropTypes.array.isRequired
    }

    componentDidMount() {
        this.props.fetchPosts()
        let elems = document.querySelectorAll('.dropdown-trigger');
        // eslint-disable-next-line
        let instances = M.Dropdown.init(elems, { alignment: 'right', constrainWidth: false })
    }

    render() {
        return (
            <div>
                <div className="category-title">
                    Showing All Posts
                <OrderPosts />
                </div>
                <div className="divider"></div>
                <PostList posts={this.props.posts} />
            </div>
        )
    }
}

const mapStateToProps = (state) => {
    return {
        posts: state.postsReducer.posts,
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        fetchPosts: () => dispatch(fetchPosts())
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(PostsPage)
