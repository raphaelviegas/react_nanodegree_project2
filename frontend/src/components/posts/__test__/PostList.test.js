import React from 'react'
import { shallow } from 'enzyme'
import { PostList } from '../PostList'
import { posts, initialState } from '../../../__test-helpers__/posts'

describe('<PostList />', () => {
    const setup = {
        ...initialState,
        fetchRemovePost: jest.fn()
    }
    let wrapper
    beforeEach(() => {
        setup.fetchRemovePost.mockClear()
        wrapper = shallow(<PostList {...setup} />)
    })

    it('Should render a message when there are no posts', () => {
        expect(wrapper.find('.noPosts')).toHaveLength(1)
        expect(wrapper).toMatchSnapshot()
    })

    it('Should render two PostCard elements', () => {
        wrapper.setProps({ posts })
        expect(wrapper.find('PostCard')).toHaveLength(2)
        expect(wrapper).toMatchSnapshot()
    })

    it('Should call fetchRemovePosts when method removePost is called', () => {
        expect(setup.fetchRemovePost).toHaveBeenCalledTimes(0)
        wrapper.instance().removePost()
        expect(setup.fetchRemovePost).toHaveBeenCalledTimes(1)
    })
})