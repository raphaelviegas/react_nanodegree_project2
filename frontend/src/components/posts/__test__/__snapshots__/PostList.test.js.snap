// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostList /> Should render a message when there are no posts 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PostList
    fetchRemovePost={[Function]}
    invalidPost={false}
    loadingPosts={true}
    posts={Array []}
    sortByOption="-voteScore"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="noPosts"
>
        <h4>
                No posts to show
        </h4>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h4>
          No posts to show
</h4>,
        "className": "noPosts",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "No posts to show",
        },
        "ref": null,
        "rendered": "No posts to show",
        "type": "h4",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="noPosts"
>
          <h4>
                    No posts to show
          </h4>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h4>
            No posts to show
</h4>,
          "className": "noPosts",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "No posts to show",
          },
          "ref": null,
          "rendered": "No posts to show",
          "type": "h4",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`<PostList /> Should render two PostCard elements 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PostList
    fetchRemovePost={[Function]}
    invalidPost={false}
    loadingPosts={true}
    posts={
        Array [
            Object {
              "author": "thingtwo",
              "body": "Everyone says so after all.",
              "category": "react",
              "commentCount": 2,
              "deleted": false,
              "id": "8xf0y6ziyjabvozdd253nd",
              "timestamp": 1467166872634,
              "title": "Udacity is the best place to learn React",
              "voteScore": 10,
            },
            Object {
              "author": "thingone",
              "body": "Just kidding. It takes more than 10 minutes to learn technology.",
              "category": "redux",
              "commentCount": 0,
              "deleted": false,
              "id": "6ni6ok3ym7mf1p33lnez",
              "timestamp": 1468479767190,
              "title": "Learn Redux in 10 minutes!",
              "voteScore": -5,
            },
          ]
    }
    sortByOption="-voteScore"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="row"
>
        <PostCard
                onRemovePost={[Function]}
                post={
                        Object {
                                "author": "thingtwo",
                                "body": "Everyone says so after all.",
                                "category": "react",
                                "commentCount": 2,
                                "deleted": false,
                                "id": "8xf0y6ziyjabvozdd253nd",
                                "timestamp": 1467166872634,
                                "title": "Udacity is the best place to learn React",
                                "voteScore": 10,
                              }
                }
        />
        <PostCard
                onRemovePost={[Function]}
                post={
                        Object {
                                "author": "thingone",
                                "body": "Just kidding. It takes more than 10 minutes to learn technology.",
                                "category": "redux",
                                "commentCount": 0,
                                "deleted": false,
                                "id": "6ni6ok3ym7mf1p33lnez",
                                "timestamp": 1468479767190,
                                "title": "Learn Redux in 10 minutes!",
                                "voteScore": -5,
                              }
                }
        />
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <PostCard
            onRemovePost={[Function]}
            post={
                        Object {
                                    "author": "thingtwo",
                                    "body": "Everyone says so after all.",
                                    "category": "react",
                                    "commentCount": 2,
                                    "deleted": false,
                                    "id": "8xf0y6ziyjabvozdd253nd",
                                    "timestamp": 1467166872634,
                                    "title": "Udacity is the best place to learn React",
                                    "voteScore": 10,
                                  }
            }
/>,
          <PostCard
            onRemovePost={[Function]}
            post={
                        Object {
                                    "author": "thingone",
                                    "body": "Just kidding. It takes more than 10 minutes to learn technology.",
                                    "category": "redux",
                                    "commentCount": 0,
                                    "deleted": false,
                                    "id": "6ni6ok3ym7mf1p33lnez",
                                    "timestamp": 1468479767190,
                                    "title": "Learn Redux in 10 minutes!",
                                    "voteScore": -5,
                                  }
            }
/>,
        ],
        "className": "row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "onRemovePost": [Function],
            "post": Object {
              "author": "thingtwo",
              "body": "Everyone says so after all.",
              "category": "react",
              "commentCount": 2,
              "deleted": false,
              "id": "8xf0y6ziyjabvozdd253nd",
              "timestamp": 1467166872634,
              "title": "Udacity is the best place to learn React",
              "voteScore": 10,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "onRemovePost": [Function],
            "post": Object {
              "author": "thingone",
              "body": "Just kidding. It takes more than 10 minutes to learn technology.",
              "category": "redux",
              "commentCount": 0,
              "deleted": false,
              "id": "6ni6ok3ym7mf1p33lnez",
              "timestamp": 1468479767190,
              "title": "Learn Redux in 10 minutes!",
              "voteScore": -5,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row"
>
          <PostCard
                    onRemovePost={[Function]}
                    post={
                              Object {
                                        "author": "thingtwo",
                                        "body": "Everyone says so after all.",
                                        "category": "react",
                                        "commentCount": 2,
                                        "deleted": false,
                                        "id": "8xf0y6ziyjabvozdd253nd",
                                        "timestamp": 1467166872634,
                                        "title": "Udacity is the best place to learn React",
                                        "voteScore": 10,
                                      }
                    }
          />
          <PostCard
                    onRemovePost={[Function]}
                    post={
                              Object {
                                        "author": "thingone",
                                        "body": "Just kidding. It takes more than 10 minutes to learn technology.",
                                        "category": "redux",
                                        "commentCount": 0,
                                        "deleted": false,
                                        "id": "6ni6ok3ym7mf1p33lnez",
                                        "timestamp": 1468479767190,
                                        "title": "Learn Redux in 10 minutes!",
                                        "voteScore": -5,
                                      }
                    }
          />
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <PostCard
              onRemovePost={[Function]}
              post={
                            Object {
                                          "author": "thingtwo",
                                          "body": "Everyone says so after all.",
                                          "category": "react",
                                          "commentCount": 2,
                                          "deleted": false,
                                          "id": "8xf0y6ziyjabvozdd253nd",
                                          "timestamp": 1467166872634,
                                          "title": "Udacity is the best place to learn React",
                                          "voteScore": 10,
                                        }
              }
/>,
            <PostCard
              onRemovePost={[Function]}
              post={
                            Object {
                                          "author": "thingone",
                                          "body": "Just kidding. It takes more than 10 minutes to learn technology.",
                                          "category": "redux",
                                          "commentCount": 0,
                                          "deleted": false,
                                          "id": "6ni6ok3ym7mf1p33lnez",
                                          "timestamp": 1468479767190,
                                          "title": "Learn Redux in 10 minutes!",
                                          "voteScore": -5,
                                        }
              }
/>,
          ],
          "className": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "class",
            "props": Object {
              "onRemovePost": [Function],
              "post": Object {
                "author": "thingtwo",
                "body": "Everyone says so after all.",
                "category": "react",
                "commentCount": 2,
                "deleted": false,
                "id": "8xf0y6ziyjabvozdd253nd",
                "timestamp": 1467166872634,
                "title": "Udacity is the best place to learn React",
                "voteScore": 10,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "onRemovePost": [Function],
              "post": Object {
                "author": "thingone",
                "body": "Just kidding. It takes more than 10 minutes to learn technology.",
                "category": "redux",
                "commentCount": 0,
                "deleted": false,
                "id": "6ni6ok3ym7mf1p33lnez",
                "timestamp": 1468479767190,
                "title": "Learn Redux in 10 minutes!",
                "voteScore": -5,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
